### ISSUE TITLE
Prevent instant death not working on dedicated server

### ISSUE THREAD

**Issue by saildeep** [acemod/ACE3#3538]
**Arma 3 Version:** `1.56.134787` stable
**CBA Version:** `2.3.1` stable
**ACE3 Version:** `3.5.0` stable

**Mods:**
- `@CBA_A3`
- `@ace`
- `@tfar`
- All CUP Mods
- CUP ACE3 Compabilty Mods 

**Description:**
The "prevent instant death" feature is not working on my dedicated server. I tried configuring the medical system via Description.ext and now moved to serverconfig.hpp. When using the description.ext everything works fine on localy hosted MP, but players may die instantly on dedicated server.

**Steps to reproduce:**
- Load the mission from the zip-file
  [bugreport_prevent_instant_death.zip](https://github.com/acemod/ACE3/files/159755/bugreport_prevent_instant_death.zip)
  on a dedicated server with the given serverconfig.hpp
- Get some money by executing "50000 call smm_add_money"
- Buy some magazines at the "gear_box" until a frag grenade appears
- Throw the grenade on the ground and wait until it explodes

**Where did the issue occur?**
- Dedicated

**Placed Modules:**
Serverconfig:
http://pastebin.com/raJv1HzB

**RPT log file:**
http://pastebin.com/DPMt5VLq

**Comment by Glowbal** [comment]
Hi @BeerRush, I have edited your post and placed the server config and rpt log file into pastebin. That makes it a bit easier to read the post!

You say:
> players may die instantly on dedicated server.

Does this means that sometimes it does work? Meaning you observe unreliable functionality from prevent instant death?
Your repo steps only talk about grenades. Is this the only case were it does this?

**Comment by kerckasha** [comment]
prevent instant death basically protects you from a singular hit, so in the case of grenades usually the explosion will kill you - prevent instant death kicks in and then you get hit by the shrapnel and subsequently die.

i personally added an invulnerability period to avoid this sort of thing in my own branched version

**Comment by BeerRush** [comment]
@Glowbal thank you for moving the files to pastebin.
 i'm currently testing. 
Shooting on the ground with a M136  or jumping from a tower works and knocks me unconscious.
@kerckasha i think you are right with this. Is there a possibility to use this in a mission without modifying ace? 

It would be nice to see this integrated in a future version of ace, because this is not only a problem with frag grenades but also machine guns etc.

**Comment by Grizmatki** [comment]
Can confirm that this is still happening. Syncing the revive setting to the medical settings has not helped at all. This is an ongoing issue and extremely frustrating.

**Comment by commy2** [comment]
Can you give any details? There are scripts in e.g. RHS that will kill you every time (`setDamage 1`) and there is nothing we can do about.

**Comment by Grizmatki** [comment]
For instance if I get shot by a Leights Opfor ISTS/Afghan Militia etc, which use RHS weapons we get instant killed. If I use a RHS M4 on a player (player to player) it works properly and prevents instant death. 

Let me know what other information you need!!

**Comment by J-Ziegler** [comment]
This being the case if you shoot someone with a single round? If the first one knocks them unconscious (rather, prevent insta-death kicks in) any subsequent hits will kill them, as mentioned above.

**Comment by Grizmatki** [comment]
Since my thread is closed. What progress was made by the ACE team?

Clearly there is a confliction with ACE and another mod. Since the developers have stopped replying to this thread it must be fixed?

What can we do to get a working ACE on our server again?

**Comment by commy2** [comment]
Your issue was closed because it was a duplicate. There is no point in leaving open duplicate issues, so please don't create them.

>Clearly there is a confliction with ACE and another mod.
So are saying this problem does not occur with only CBA + ACE on a newly created mission? Do I understand you correctly?

If you want to help diagnosing this problem you should:

- figure out what third party addon exactly causes this problem.
- upload the mission and RPT file
- enable the debug console, join the server and report to me what this shows in the chat if exected __once__ (LOCAL EXEC):

```
_ehid = player addEventHandler ["handleDamage", {}];
systemChat str _ehid;
copyToClipboard str _ehid;
```

**Comment by Grizmatki** [comment]
I honestly don't have time to do testing of broken mods. You're asking me to sort through 40+ mods when multiple people have reported this issue. I understand if I was the only one. 

But as you even said there are duplicates, which means others with different mod sets are being affected.

Now if you looked at my thread and compare it to this thread before you closed it you would have realised that its evident that there are different mods causing it. The only similar mods are ACE, CBA, AND CUP CORE/Terrains. 

Please don't waste my time and I wont waste yours. You are asking me to do work that is already done for you through your report a bug feature.

**Comment by BaerMitUmlaut** [comment]
>I honestly don't have time to do testing of broken mods

Good for you, we don't either. The problem is in one of the mods of your modpack. I'd suggest you give us the required information or you learn to live with your problem.

**Comment by Grizmatki** [comment]
>  I'd suggest you give us the required information or you learn to live with your problem.

Let me understand. If ACE and another mod don't work together. We are responsible for finding and doing all the testing? Even ALiVE's dev team was more willing to help out their users!

**Comment by BaerMitUmlaut** [comment]
We do neither have the resources nor the time to test every single one of the thousands of Arma 3 Mods there are together with ACE3. Especially not with some severely outdated mods that you are using (for example, the EODS mod, which has last been updated 2014 and never left the Alpha stage).

Instead, we require at least some sort of indication what causes the problem. Step one would be an RPT file, which takes you 5 minutes to upload. Step two in this case would be executing the code commy provided. This takes you 5 minutes to do, and another 5 minutes to upload a screenshot or copy the output here. Total time: **15 minutes**. That is far less than the time this discussion has already taken.

If you do not have these 15 minutes, then I don't suppose your problem is that much of a big deal for you.

**Comment by Grizmatki** [comment]
EODS has been updated multiple times and the last update was 5 days ago, please check their armaholic page in case you were out of the loop!  (The Burns 15th Edit FYI).

I have 10+ 3 hours long operation RPT files, I will upload all of them if you wish.

But testing and singling out each mod, upload the mission to the server? Thats going to take 5-6 hours, not 15 minutes like you are suggesting.

**Comment by Grizmatki** [comment]
[Link to dropbox with 3+ hour long operations](https://www.dropbox.com/sh/ez8u9q1d0fwamb0/AABojOIAJKmIZYJv32Pd9wSxa?dl=0)

**Comment by BaerMitUmlaut** [comment]
Thank you for the RPTs.
Yes, testing with each mod is certainly going to take more time. We'll see if we can find the error like this, or if we have to resort to more time intensive methods. Can you post the output of the code commy posted as well?

**Comment by Grizmatki** [comment]
Let me understand how you want the code pasted first. I just ran it on my server. You would like me to set up the situation, and then before I get enganged you want me to run the code and paste the result immediately after I die?

**Comment by BaerMitUmlaut** [comment]
Run it about a minute after mission start or later.

**Comment by Grizmatki** [comment]
Running it a minute after mission start just gives me.

4 in the chat box.

**Comment by Grizmatki** [comment]
Just tested this out. Got shot by a ISTS with AK-47 and prevent instant death worked. Got shot twice by a lee enfield and the second shot killed me. Was never knocked out, just injured by the first shot (shot in the arm). Prevent instant death is completely random.

**Comment by BaerMitUmlaut** [comment]
There we go, the chat message told us almost certainly the cause of the problem. An addon is trying to handle the player damage and is overwriting ACE3's damage handler. I'll let you know what to do next, but it will have to wait until tomorrow, it's 1:37am here.

**Comment by Grizmatki** [comment]
Alright I'll look into it some more. I'll start with addons that affect player/armor damage and see what I can find.

**Comment by Grizmatki** [comment]
What number are we looking for 2/3?

**Comment by PabstMirror** [comment]
Ace_medical should add Index 0
Arma 3 will add Index 1

So adding another should return 2

**Comment by Grizmatki** [comment]
So to confirm, when we input the script. We should get a return value of 2?  Or 3 because the script "adds" an event handler?

**Comment by commy2** [comment]
2 would be correct. Also note that the mission can add `handleDamage` eventhandlers too, which is why I asked you to upload the mission file.
On the other hand, not every `handleDamage` eventhandler would be a porblem. Only those returning numbers to overwrite the damage the player has taken.
If my code reports 4 for you, that means that there are at least 2 additional `handleDamage` eventhandlers on soldier objects. If any of these sets the player total damage to 1 or higher, the game will kill you, no matter what we as ACE team do, because you told the game to do so.

**Comment by PabstMirror** [comment]
This is quite hacky, but you could try something like this:
```
[] spawn {
    sleep 1;
    {
        _x removeAllEventHandlers "HandleDamage";
        _x addEventHandler ["HandleDamage", ace_medical_fnc_handleDamage];
    } forEach allUnits;
};
```
This code would have issues with new spawned units/jips, but for a simple test it should show if the issues are caused by the extra event handlers.

**Comment by Glowbal** [comment]
Will be done after #3916 and #3917

**Comment by kamkill01011** [comment]
@kerckasha How do you add an invulnerability period to prevent instant death ?

**Comment by ace3mod** [comment]
Hello @acemod/maintainers. This ticket has been open for over 168 days without any activity.
